File: JavaScript/b-call-apply.js
'use strict';

function f1(a, b) {
  console.log('f1(' + a + ', ' + b + ')');
}

f1(2, 3);
f1.call(null, 2, 3);

const arr = [2, 3];
f1(...arr);
f1.apply(null, arr);

File: Ruby/b-call-apply.rb
=begin
Example aplly Unrelated methods 
that can be called only after they 
are associated with an object
=end
class Square
  def area
    @side * @side
  end
  def initialize(side)
    @side = side
  end
end

area_un = Square.instance_method(:area)

s = Square.new(12)
area = area_un.bind(s)
puts area.call   #=> 144




def f1(a,b)
    puts "f1(#{a}, #{b})"
end

arr = [1,2]

f1(1,2)

f1(*arr)
